builtin/credential/cert/path_login.go

```diff
@@ -609,6 +616,10 @@ func (b *backend) loadTrustedCerts(ctx context.Context, storage logical.Storage,
 	trustedCerts = make([]*ParsedCert, 0)
 	trustedNonCAs = make([]*ParsedCert, 0)
 
+	lock := locksutil.LockForKey(b.trustedCacheLocks, certName)
+	lock.Lock()
+	defer lock.Unlock()
+
```

Comment:
To avoid re-doing the work, shouldn't we check to make sure that `b.trustedCache.Get(certName)` keep doing the work only if nil, otherwise leverage the newly populated cache while we were locked?

Reply:
You only get to this point after a cache test in getTrustedCerts, so yeah, there is a tiny race here where it could have been filled in prior to getting the lock.  Worth another test?

Reply:
Well the point of the lock was to avoid different clients from re-populating the cache? All this will do is actually make it worse by serializing all the requests but they will still re-populate the cache no?

Reply:
Doh!  You're right.  I should take an RLock in the read case.  And sure, I'll do a re-test.